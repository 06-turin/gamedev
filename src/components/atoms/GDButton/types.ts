import { ButtonHTMLAttributes } from 'react';

export type GDButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
    title: string
    styleOption: 'primary' | 'secondary'
}
