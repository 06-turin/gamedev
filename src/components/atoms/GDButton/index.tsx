export { GDButtonComponent as GDButton } from './GDButton';
