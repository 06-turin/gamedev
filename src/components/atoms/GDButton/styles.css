.btn {
  box-shadow: none;
  margin-top: 18px;
  color: var(--white);
  background-color: transparent;
  text-transform: uppercase;

  &.size_h {
    font-size: 96px;
    padding: 8px 22px 6px 22px;
    border-radius: 12px;
  }

  &.size_l {
    font-size: 32px;
    padding: 8px 22px 6px 22px;
    border-radius: 12px;
  }

  &.size_m {
    font-size: 24px;
    padding: 8px 22px 6px 22px;
    border-radius: 12px;
  }

  &.size_s {
    font-size: 18px;
    padding: 4px 12px;
    border-radius: 6px;
  }

  &-primary {
    border: 2px solid var(--white);

    &:hover {
      background-color: var(--white);
      border-color: var(--activityColor);
      color: var(--appBackground);
    }

    &:active {
      background-color: var(--activityColor);
      border-color: var(--activityColor);
      color: var(--appBackground);
    }
  }

  &-secondary {
    border: none;

    &:hover {
      color: var(--activityColor);
    }
  }

  &:hover {
    cursor: pointer;
  }
}
