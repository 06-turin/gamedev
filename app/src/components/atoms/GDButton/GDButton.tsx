import './styles.css';
import React, { FC, MouseEventHandler } from 'react';
import classNames from 'classnames';
import { GDButtonStyleOption, GDSizeOption } from 'components/organisms/App/types';

export type GDButtonProps = {
  title: string
  styleOption?: GDButtonStyleOption
  onClick?: MouseEventHandler<HTMLButtonElement>
  className?: string
  size?: GDSizeOption
  type?: 'submit' | 'reset' | 'button'
  disabled?: boolean
}

export const GDButton: FC<GDButtonProps> = ({
  title,
  onClick,
  styleOption = 'primary',
  className,
  size = 'm',
  type = 'button',
  disabled = false,
}) => (
  <button
    type={type}
    onClick={onClick}
    disabled={disabled}
    className={classNames(['btn', `btn-${styleOption}`, `size_${size}`, className])}
  >
    {title}
  </button>
);
