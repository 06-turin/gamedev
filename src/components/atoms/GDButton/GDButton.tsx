import './styles.css';
import React, { FC, MouseEventHandler } from 'react';
import classNames from 'classnames';

export type GDButtonProps = {
  title: string
  styleOption: 'primary' | 'secondary'
  onClick: MouseEventHandler<HTMLButtonElement>
  className?: string
  size?: 's' | 'm' | 'l' | 'h',
  type?: 'submit' | 'reset' | 'button';
}

export const GDButton: FC<GDButtonProps> = ({
  title,
  onClick,
  styleOption,
  className,
  size = 'm',
  type = 'button',
}) => (
  <button
    type={type}
    onClick={onClick}
    className={classNames(['btn', `btn-${styleOption}`, `size_${size}`, className])}
  >
    {title}
  </button>
);
