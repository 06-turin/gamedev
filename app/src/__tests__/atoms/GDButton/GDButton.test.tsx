import React from 'react';
import classNames from 'classnames';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { GDButton } from 'components/atoms/GDButton/GDButton';
import '@testing-library/jest-dom/extend-expect';

const mockFunction = () => 'returns stuff';

describe('GDButton', () => {
  it('renders with default props', () => {
    const tree = renderer
      .create(
        <GDButton title="title" />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with all props passed', () => {
    const tree = renderer
      .create(
        <GDButton
          title="title"
          onClick={mockFunction}
          styleOption="secondary"
          size="l"
          type="submit"
          className={classNames('extra-class-name')}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with the right title', () => {
    const { getByText } = render(<GDButton title="title being checked" />);
    expect(getByText('title being checked')).toBeInTheDocument();
  });
});
