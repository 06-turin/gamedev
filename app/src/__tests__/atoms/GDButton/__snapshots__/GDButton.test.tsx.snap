// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GDButton renders with all props passed 1`] = `
<button
  className="btn btn-secondary size_l extra-class-name"
  onClick={[Function]}
  type="submit"
>
  title
</button>
`;

exports[`GDButton renders with default props 1`] = `
<button
  className="btn btn-primary size_m"
  type="button"
>
  title
</button>
`;
